tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py

```diff
@@ -405,3 +400,37 @@ def _get_default_auxiliary_params(self) -> dict:
         )
         default_auxiliary_params.update(extra_auxiliary_params)
         return default_auxiliary_params
+
+    def __get_metrics_map(self):
+        from fastai.metrics import rmse, mse, mae, accuracy, FBeta, RocAucBinary, Precision, Recall, R2Score
+        metrics_map = {
+            # Regression
+            'root_mean_squared_error': rmse,
+            'mean_squared_error': mse,
+            'mean_absolute_error': mae,
+            'r2': R2Score(),
+            # Not supported: median_absolute_error
```

Comment:
since eval_metric is mainly just used for early-stopping, maybe if a metric is not supported, you can still map it to the most similar metric for training the FastAI net?  Eg. for median_absolute_error, you could choose mae. And you could print low-priority output debug message that:  training objective for FastAI net is actually mean_absolute_error instead of median_absolute_error because the latter is not supported as a training objective.

Reply:
I ended up implementing it myself - update PR.